### PR TITLE
OSS::InformationSection component

### PR DESCRIPTION
**Pull request by OwenCoogan**
### What does this PR do?

It adds a new component used in the 2FA project

Capture d’écran 2024-12-16 à 15 44 32
Capture d’écran 2024-12-16 à 15 44 13
Capture d’écran 2024-12-16 à 15 36 56
Capture d’écran 2024-12-16 à 12 05 44

Related to: #

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [ ] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/information-section.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import Component from '@glimmer/component';
+import type { IconNames } from './iconName.enum';
+
+interface InformationSectionArgs {
+  imageBadge: keyof typeof IconNames | string;
```
Comment (nathalieArnoux) [inline addon/components/o-s-s/information-section.ts:5]: ? :thinking:

File: tests/integration/components/o-s-s/information-section-test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  });
+
+  test('it renders the @subtitle arg', async function (assert) {
+    await render(hbs`<OSS::InformationSection @title={{this.title}} @subtitle={{this.subtitle}}/>`);
+
+    assert.dom("[data-control-name='information-section--subtitle']").hasText(this.subtitle);
+  });
+
+  module('plain arg', function () {
+    test('it renders the header with a grey background when @plain is false', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/information-section-test.ts:41]: nit: maybe add a test for the default behavour, when @plain is not specified

File: tests/integration/components/o-s-s/information-section-test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    });
+  });
+
+  module('content named block', function () {
+    test('it renders the badge with content named block', async function (assert) {
+      await render(hbs` <OSS::InformationSection @title={{this.title}}>
+                          <:content>
+                          <p data-control-name='content-named-block'>{{this.namedBlockContent}}</p>
+                          </:content>
+                        </OSS::InformationSection>`);
```
Comment (Miexil) [inline tests/integration/components/o-s-s/information-section-test.ts:104]: nit: alignment of the <p> tag
It also looks like you have an empty state after each `await render` statement in some of the tests of the file

File: addon/components/o-s-s/information-section.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import Component from '@glimmer/component';
+import type { IconNames } from './iconName.enum';
+
+interface InformationSectionArgs {
+  badgeIcon?: keyof typeof IconNames;
+  svgIcon?: string;
```
Comment (Miexil) [inline addon/components/o-s-s/information-section.ts:6]: `imageUrl` should be a bit more "permissive" language wise :) 
the icons might be pngs sometimes

File: app/styles/atoms/information-section.less [hunk 5]
```diff
@@ -0,0 +1,10 @@
+.oss--information-section {
```
Comment (Miexil) [inline app/styles/atoms/information-section.less:1]: please space everything properly (missing line breaks)

File: addon/components/o-s-s/information-section.stories.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import { hbs } from 'ember-cli-htmlbars';
+
+export default {
+  title: 'Components/OSS::InformationSection',
+  argTypes: {
+    title: {
```
Comment (Miexil) [inline addon/components/o-s-s/information-section.stories.js:6]: you can be a little more precise with the argTypes in storybook, there are examples in other files in the project
```javascript
      table: {
        type: { summary: 'string' },
        defaultValue: { summary: 'undefined' }
      },
      control: {
        type: 'text'
      }
```

File: addon/components/o-s-s/information-section.stories.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+      @title={{this.title}}
+      @badgeIcon={{this.badgeIcon}}
+      @subtitle={{this.subtitle}}
+      @plain={{this.plain}}
+    >
+     <:action>
+      <div class="fx-row fx-gap-px-6">
+        <OSS::Tag @label="Hello" @skin="primary" />
+        <OSS::Button @size="sm" @label="Action Button"/>
+      </div>
```
Comment (Miexil) [inline addon/components/o-s-s/information-section.stories.js:81]: nit: indentation

File: addon/components/o-s-s/information-section.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import Component from '@glimmer/component';
+import type { IconNames } from './iconName.enum';
+
+interface InformationSectionArgs {
+  badgeIcon?: keyof typeof IconNames;
```
[the remark this answers is not in the document]
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.ts:5]: Yep 
Capture d’écran 2024-12-17 à 10 26 10

File: addon/components/o-s-s/information-section.stories.js (hunk 7 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.stories.js]: Fixed indentation

File: addon/components/o-s-s/information-section.stories.js (hunk 6 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.stories.js:6]: Specified argTypes in sb

File: app/styles/atoms/information-section.less (hunk 5 above)
Comment (OwenCoogan) [reply app/styles/atoms/information-section.less:1]: proper cleanup

File: addon/components/o-s-s/information-section.ts (hunk 4 above)
Comment (OwenCoogan) [reply addon/components/o-s-s/information-section.ts]: Changed to imageUrl

File: tests/integration/components/o-s-s/information-section-test.ts (hunk 3 above)
Comment (OwenCoogan) [reply tests/integration/components/o-s-s/information-section-test.ts]: Fixed testfile

File: tests/integration/components/o-s-s/information-section-test.ts (hunk 2 above)
Comment (OwenCoogan) [reply tests/integration/components/o-s-s/information-section-test.ts:38]: Added New test

**Review by nathalieArnoux — APPROVED** [review]

File: addon/components/o-s-s/information-section.stories.js (hunk 6 above)
Comment (Miexil) [reply addon/components/o-s-s/information-section.stories.js:6]: I think the `defaultValue`s are wrong ^^
should be 'undefined' everywhere no ? 🤷

**Review by Miexil — APPROVED** [review]

**Pull request merged**